[Tests][CI] contrib LaTeX tests fail instead of skipping when dependencies missing

## Describe the issue

When running ```python -m pytest cirq-core/``` some tests in cirq.contrib failed when LaTeX dependencies are missing or partially installed, instead of being skipped. The error is (when grep with error/warning string the outputs):
```
cirq-core/cirq/contrib/quantikz/circuit_to_latex_render_test.py:42: AssertionError
stdout = b'Latexmk: This is Latexmk, John Collins, 26 Dec. 2019, version: 4.67.\nRule \'latex\': The following rules & subrules... or warnings treated as errors.\n=== TeX engine is \'pdfTeX\'\nLatexmk: Errors, so I did not complete making targets\n'
        CalledProcessError. The CalledProcessError object will have the return code
        triggered by setting any of text, encoding, errors or universal_newlines.
                raise ValueError('stdin and input arguments may not both be used.')
                raise ValueError('stdout and stderr arguments may not be used '
>               raise CalledProcessError(retcode, process.args,
E               subprocess.CalledProcessError: Command '['latexmk', '-pdfps', '--interaction=nonstopmode', '/tmp/pytest-of-ubuntu/pytest-38/test_qcircuit_pdf0/test_file.tex']' returned non-zero exit status 12.
/usr/lib/python3.11/subprocess.py:571: CalledProcessError
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
! LaTeX Error: Missing \begin{document}.
Collected error summary (may duplicate other messages):
 unless error was exceeding maximum runs, or warnings treated as errors.
Latexmk: Errors, so I did not complete making targets
...
FAILED cirq-core/cirq/contrib/quantikz/circuit_to_latex_render_test.py::test_render_circuit
FAILED cirq-core/cirq/contrib/qcircuit/qcircuit_pdf_test.py::test_qcircuit_pdf
= 2 failed, 20025 passed, 10 skipped, 64 xfailed, 21 warnings in 289.77s (0:04:49) =
```

Some documentation in some deep subfolder helps to resolve it, but it takes some effort to find it:
```
cirq-core/cirq/contrib/quantikz/circuit_to_latex_render.py:            "On Ubuntu/Debian: `sudo apt-get install texlive-full` "
```
However, it will be a good idea to add it in the main README.

Nonetheless, **the best practice** is likely to skip when one of the LaTeX/TeX dependencies is missing, adding something like this: 
```
pytest.mark.skipif(shutil.which("pdflatex") is None, ...)
```

## Tell us the version of Cirq where this happens

I use Cirq from source version: ```commit 9888067c5c36b4505917de57f0528c53882ec240```
Python: Python 3.11.11

You can get the Cirq version you are running by using Python
to print the value of `cirq.__version__`. From the command line:

```shell
python -c 'import cirq; print(cirq.__version__)'
```
This prints:
```
(.venv) ubuntu@fuzzer-03:~/Cirq$ python -c 'import cirq; print(cirq.__version__)'
1.6.1
```

--- a/.github/workflows/ci-monitoring.yml
+++ b/.github/workflows/ci-monitoring.yml
@@ -42,7 +42,7 @@ jobs:
       - name: Install system packages
         run: |
           sudo apt-get update
-          sudo apt-get install latexmk texlive-latex-base texlive-latex-extra
+          sudo apt-get install $(cat apt-system-requirements.txt)
       - name: Install requirements
         run: |
           pip install --upgrade setuptools wheel

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -102,7 +102,7 @@ jobs:
       - name: Install system packages
         run: |
           sudo apt-get update
-          sudo apt-get install latexmk texlive-latex-base texlive-latex-extra
+          sudo apt-get install $(cat apt-system-requirements.txt)
       - name: Install changed files test dependencies
         run: dev_tools/conf/pip-install-minimal-for-pytest-changed-files.sh
       - name: Changed files test
@@ -215,7 +215,7 @@ jobs:
       - name: Install system packages
         run: |
           sudo apt-get update
-          sudo apt-get install latexmk texlive-latex-base texlive-latex-extra
+          sudo apt-get install $(cat apt-system-requirements.txt)
       - name: Install requirements
         run: |
           pip install --upgrade setuptools wheel
@@ -279,7 +279,7 @@ jobs:
       - name: Install system packages
         run: |
           sudo apt-get update
-          sudo apt-get install latexmk texlive-latex-base texlive-latex-extra
+          sudo apt-get install $(cat apt-system-requirements.txt)
       - name: Install requirements
         run: |
           pip install --upgrade setuptools wheel

--- a/apt-system-requirements.txt
+++ b/apt-system-requirements.txt
@@ -1,3 +1,3 @@
-texlive-latex-base
 latexmk
 python3-tk
+texlive-latex-extra

